include/dfu/mcuboot.h

```diff
@@ -17,26 +17,57 @@
 extern "C" {
 #endif
 
+#ifdef BOOT_SWAP_TYPE_NONE
+#if BOOT_SWAP_TYPE_NONE != 1 /*ensure the same definition in MCUBoot */
+#error "definition incompatible"
+#endif
+#else
 /** Attempt to boot the contents of slot 0. */
 #define BOOT_SWAP_TYPE_NONE     1
+#endif
 
+#ifdef BOOT_SWAP_TYPE_TEST
+#if BOOT_SWAP_TYPE_TEST != 2  /*ensure the same definition in MCUBoot */
+#error "definition incompatible"
+#endif
+#else
 /** Swap to slot 1.  Absent a confirm command, revert back on next boot. */
 #define BOOT_SWAP_TYPE_TEST     2
+#endif
 
+#ifdef BOOT_SWAP_TYPE_PERM
+#if BOOT_SWAP_TYPE_PERM != 3  /*ensure the same definition in MCUBoot */
+#error "definition incompatible"
+#endif
+#else
 /** Swap to slot 1, and permanently switch to booting its contents. */
 #define BOOT_SWAP_TYPE_PERM     3
+#endif
 
+#ifdef BOOT_SWAP_TYPE_REVERT
+#if BOOT_SWAP_TYPE_REVERT != 4  /*ensure the same definition in MCUBoot */
+#error "definition incompatible"
+#endif
+#else
 /** Swap back to alternate slot.  A confirm changes this state to NONE. */
 #define BOOT_SWAP_TYPE_REVERT   4
+#endif
 
+#ifdef BOOT_SWAP_TYPE_REVERT
+#if BOOT_SWAP_TYPE_REVERT != 4  /*ensure the same definition in MCUBoot */
```

Comment:
Is this really supposed to check for `BOOT_SWAP_TYPE_REVERT` and not `BOOT_SWAP_TYPE_FAIL`?